src/jsdoc_transformer.ts

```diff
@@ -307,6 +288,55 @@ function createMemberTypeDeclaration(
   return ts.createIf(ts.createLiteral(false), ts.createBlock(propertyDecls, true));
 }
 
+/**
+ * Declares to Closure the supertypes of a given type.
+ * Normally you'd do this with \@implements or \@extends, but this is used in
+ * cases where we cannot express the types using those tags.
+ */
+function createSuperTypeRelationship(subType: string, superTypes: string[]): ts.IfStatement {
+  // For each supertype-subtype relationship, we create an unused function that does:
+  //   function tsickleSuperType(/** !Fruit */ supertype, /** !Apple */ subtype) {
+  //     supertype = subtype;
+  //   }
+  // This is an illegal assignment because the types aren't known to be related,
+  // but because checkTypes is off(!) it instead signals to the compiler that the
+  // types should be treated as if they were related.
+  const relationStatements: ts.Statement[] = [];
+  for (const superType of superTypes) {
+    const superParam = ts.createParameter(
+        /* decorators */ undefined, /* modifiers */ undefined,
+        /* dotDotDot */ undefined, 'supertype');
+    ts.addSyntheticLeadingComment(
+        superParam, ts.SyntaxKind.MultiLineCommentTrivia, `* !${superType} `,
```

Comment:
I've raised a thread on `js-compiler-users@` on whether this syntax is reliable. I'm always a bit spooked out by this (does `/* {foo: string} */` work? Does `/* X<A>|B|* */` work? it's so hard to tell...). I think it's fine to go with this for the time being, but you might also consider re-using our support for JSDoc and `@param` and serializing those, just for consistency.